registry/config.go

```diff
@@ -143,13 +149,24 @@ func (config *serviceConfig) LoadInsecureRegistries(registries []string) error {
 
 skip:
 	for _, r := range registries {
-		// validate insecure registry
 		if _, err := ValidateIndexName(r); err != nil {
 			// before returning err, roll back to original data
 			config.ServiceConfig.InsecureRegistryCIDRs = originalCIDRs
 			config.ServiceConfig.IndexConfigs = originalIndexInfos
 			return err
 		}
+		// validate insecure registry
+		if schemeRegex.MatchString(r) {
+			// Remove https?:// and warn
+			logrus.Warnf("insecure registry %s should not contain '%s://'", r, strings.Split(r, "://")[0])
```

Comment:
The warning should make it clear that the insecure registry option is still being honored.